File: program_data/github_java_sort_function/8/103.java
   
    public void quicksort(int left, int right) {
        int l1 = left, r1 = right;

        if (left > right)
            return;

        while (l1 != r1) {
            
            while (b[r1] >= b[left]&&l1<r1) {
                r1--;
            }
            while (b[l1] <= b[left]&&l1<r1) {
                l1++;
            }
            
            if (l1 < r1) {
                int temp = b[r1];
                b[r1] = b[l1];
                b[l1] = temp;
            }
        }

        
        
        
        
        int temp = b[left];
        b[left] = b[l1];
        b[l1] = temp;
        
        quicksort(left, l1 - 1);
        
        quicksort(l1+1, right);
    }

File: program_data/PKU_raw/49/103.c
//*************************************************
//* file: 1000012912_005.cpp                      *
//* author : ???                               *
//* date : 2010.11.26                             *
//* function : ????                           *
//*************************************************
void huiwen(int pos, int l , char str[]);
int main()
{
	int i, len, l;                                        //l?????
	char str[500];  
	cin.getline(str,500);
	len = strlen(str);
	for ( l = 2; l <= len; l ++ )
	{
		for ( i = 0; i < len; i ++ )                     //i?????
			 huiwen(i,l,str);
	}
	return 0;
}
void huiwen( int pos, int l, char str[])
{
	int right, left, j;                                                          //left?????right????
	for ( left = pos, right = pos + l - 1; right > left; right --, left ++ )
		if ( str[left] != str[right] )                                          //?????????
			break;
	if ( left >= right )                                     //????
	{
		for ( j = pos; j < pos + l; j ++ )
			cout << str[j];
		cout << endl;
	}
}